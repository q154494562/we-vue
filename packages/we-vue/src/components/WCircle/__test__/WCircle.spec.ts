import { shallowMount } from '@vue/test-utils'
import Circle from '../'

describe('circle', () => {
  test('create', () => {
    const wrapper = shallowMount(Circle, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-circle')
    expect(wrapper.classes()).toContain('wv-circle')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('what value change', () => {
    const wrapper = shallowMount(Circle, {
      propsData: {
        value: 0,
      },
    })

    wrapper.setProps({
      value: 100,
    })

    expect(wrapper.vm.currentValue).toBe(100)
  })

  test('what currentValue change', () => {
    const wrapper = shallowMount(Circle, {
      propsData: {
        value: 0,
      },
    })

    wrapper.setData({
      currentValue: 100,
    })

    expect(wrapper.emitted().input).toBeTruthy()
  })
})
