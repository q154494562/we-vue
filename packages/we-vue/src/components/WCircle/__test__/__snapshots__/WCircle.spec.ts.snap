// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle create 1`] = `
<div class="wv-circle" style="width: 100px; height: 100px;"><svg width="100" height="100" viewBox="0 0 100 100">
    <path d="M 50,50 m 0,-48
      a 48,48 0 1 1 0,96
      a 48,48 0 1 1 0,-96" stroke="#D9D9D9" stroke-width="4" fill="none"></path>
    <path d="M 50,50 m 0,-48
      a 48,48 0 1 1 0,96
      a 48,48 0 1 1 0,-96" stroke-linecap="round" stroke="#3FC7FA" stroke-width="4" fill="transparent" style="stroke-dasharray: 301.59289474462014px 301.59289474462014px; stroke-dashoffset: 301.59289474462014px; transition: stroke-dashoffset 500ms ease 0s, stroke 500ms ease;"></path><text text-anchor="middle" dominant-baseline="middle" x="50%" y="50%" class="wv-circle-content"></text>
  </svg></div>
`;
